### DISCUSSION TITLE
Why multiple EBs per stage?

### DISCUSSION THREAD

**Discussion by SupernaviX** [input-output-hk/ouroboros-leios#71] [General]
From reading the research paper, it sounds like
1. By the time EB generation begins, every relevant IB will have already been propagated through the whole chain
2. One pool can generate multiple EBs in a single stage

If we don't have to worry about late straggler IBs arriving during the link stage, why is it necessary to generate more than one EB? Are we assuming that EBs have some maximum capacity for IBs?

**Comment by pagio** [comment]
We are aiming to generate at least one *honest* EB per pipeline (with good probability). Given an honest majority of stake assumption, and the fact that block generation is probabilistic, more than one EBs per pipeline may be generated. No maximum IB capacity is assumed for EBs.

**Comment by SupernaviX** [comment]
It makes sense to generate more than one EB _per pipeline_. Do we still want individual pools to generate more than one EB (with identical contents), if they win more than one lottery in that pipeline?

**Reply by pagio** [reply]
No that would not make sense.

**Reply by SupernaviX** [reply]
Gotcha! Wasn't sure if there was some requirement that the majority of EBs were honest.

**Reply by pagio** [reply]
Not on short Leios.
